resources/pod.py

```diff
@@ -0,0 +1,158 @@
+"""
+Pod related functionalities and context info
+
+Each pod in the openshift cluster will have a corresponding pod object
+"""
+import logging
+import tempfile
+import yaml
+
+from ocs.ocp import OCP
+from ocs import defaults, kinds
+from ocsci.config import ENV_DATA
+from ocs.exceptions import CommandFailed
+
+from resources.base_resource import BaseOCSClass
+
+logger = logging.getLogger(__name__)
+
+
+class Pod(BaseOCSClass):
+    """
+    Handles per pod related context
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Initializer function
+
+        kwargs:
+            Copy of ocs/defaults.py::<some pod> dictionary
+        """
+        self.pod_data = kwargs
+        super(Pod, self).__init__(
+            self.pod_data.get('apiVersion'), self.pod_data.get('kind'),
+            self.pod_data.get('metadata').get('namespace')
+        )
+
+        self.temp_yaml = tempfile.NamedTemporaryFile(
+            mode='w+', prefix='POD_', delete=False
+        )
+        self._name = self.pod_data.get('metadata').get('name')
```

Comment:
Ok I think I solved this:

I changed reload() to:
```python
    def reload(self):
        self.data = templating.load_yaml_to_dict(self.temp_yaml.name)
        self.__init__(**self.data)
```
so it will init the obj again with the new data and by that the name will be the true name of the pod and also added call to reload() in create function:
```python
    def create(self):
        log.info(f"Adding {self.kind} with name {self.name}")
        templating.dump_dict_to_temp_yaml(self.data, self.temp_yaml.name)
        status = self.ocp.create(yaml_file=self.temp_yaml.name)
        self.reload()
        return status
```

Reply:
Could you please provide more information on why is `reload` called? And maybe add docstring to that method...

Reply:
Yes,

When you are creating a new resource, you have a simple yaml file.
After the creation, the yaml file is being updated with more information about this new resource.
For example, from the UI try to create a PV - you'll see very short yaml file.
After creation, and edit this PV, you'll see more information appended so now we also have the status, creation time etc..
That's the main reason - to stay as close as possible to the product

Reply:
@fbalak thanks for asking that - now I realized this function should have been updated.
Fixing now